transcom/mymove
# B 17951 add convert to customer expense if sit is set to expire within 30 days

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A869456)


## Summary

Added a link ("Convert to customer expense") on the "Move task order" that is enabled if SIT days remaining is 30 days or less.  If the days remaining is greater than 30, then the link will not be displayed.  When the link is clicked it opens a modal displaying SIT information and a text area to write the reason for the conversion.  The "save" button will only enable if text is entered in the text area.  

If a reason is entered and "save" is clicked, nothing happens.  The goal of the story was to create a link to a modal where the reason for conversion could be entered.  The code for the actual update will be completed in another story and then connected to the modal.  

### How to test

1. create a new move
2. add an HHG shipment
3. submit move
4. login as counselor and approve the move.
5. login as too and approve the move.
6. login as prime and add an origin SIT service item to move.
7. login as too and "accept" all origin SIT service items
8. If the "Total days remaining" is greater than 30, confirm the button "Convert to customer expense" does not appear to the left of "Edit". 
9. Click "Edit" and change "Total days of SIT approved" so that "Total days remaining" is 30 or less.  (Note: There's an existing bug that does not dynamically update "Total days remaining" as "Total days of SIT approved" is changed.  The edit has to be saved before "Total days remaining" is changed.
10. Confirm "Convert to customer expense" appears to the left of "Edit".
11. Click "Convert to customer expense" button and confirm a new modal appears labeled "Convert SIT to Customer Expense".
12. Confirm the "Office remarks" box is empty and the "save" button disabled.
13. Enter text in "Office remarks" and confirm the "save" button is enabled.

Testing complete.

## Screenshots
![image](https://github.com/transcom/mymove/assets/146897935/da16e1e1-5a8d-45d3-8533-41291480a77a)

![image](https://github.com/transcom/mymove/assets/146897935/b51f289b-87f8-41a6-8dd2-84dd6638dbf6)

![image](https://github.com/transcom/mymove/assets/146897935/3241c23b-1702-44a6-9d31-f1faea68ff1b)

![image](https://github.com/transcom/mymove/assets/146897935/3703bd6f-8b6a-4ba0-a578-8057cbef84ed)
